testing/kuttl/e2e/pgbackrest-restore/16--check-data.yaml

```diff
@@ -0,0 +1,100 @@
+        - name: psql
+          image: registry.developers.crunchydata.com/crunchydata/crunchy-postgres:centos8-14.1-0
+          env:
+            - name: PGURI
+              valueFrom: { secretKeyRef: { name: original-pguser-original, key: uri } }
+
+            # Do not wait indefinitely.
+            - { name: PGCONNECT_TIMEOUT, value: '5' }
+
+          # Note: the `$$$$` is reduced to `$$` by Kubernetes.
+          # - https://kubernetes.io/docs/tasks/inject-data-application/
+          command:
+            - psql
+            - $(PGURI)
+            - -qa
+            - --set=ON_ERROR_STOP=1
+            - --command
+            - |
+              DO $$$$
+              DECLARE
+                restored jsonb;
+              BEGIN
+                SELECT jsonb_agg(important) INTO restored FROM important;
+                ASSERT restored = '[
+                  {"data":"treasure"}, {"data":"water"}, {"data":"socks"}
+                ]', format('got %L', restored);
+              END $$$$;
+
+---
+# Confirm that replicas are also restored and streaming from the primary.
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: original-pitr-replica-data
+  labels: { postgres-operator-test: kuttl }
+spec:
+  backoffLimit: 3
+  template:
+    metadata:
+      labels: { postgres-operator-test: kuttl }
+    spec:
+      restartPolicy: Never
+      containers:
+        - name: psql
+          image: registry.developers.crunchydata.com/crunchydata/crunchy-postgres:centos8-14.1-0
+          env:
+            - name: PGPORT
+              valueFrom: { secretKeyRef: { name: original-pguser-original, key: port } }
+            - name: PGDATABASE
+              valueFrom: { secretKeyRef: { name: original-pguser-original, key: dbname } }
+            - name: PGUSER
+              valueFrom: { secretKeyRef: { name: original-pguser-original, key: user } }
+            - name: PGPASSWORD
+              valueFrom: { secretKeyRef: { name: original-pguser-original, key: password } }
+
+            # The user secret does not contain the replica service.
+            - name: NAMESPACE
+              valueFrom: { fieldRef: { fieldPath: metadata.namespace } }
+            - name: PGHOST
+              value: "original-replicas.$(NAMESPACE).svc"
+
+            # Do not wait indefinitely.
+            - { name: PGCONNECT_TIMEOUT, value: '5' }
+
+          # Note: the `$$$$` is reduced to `$$` by Kubernetes.
+          # - https://kubernetes.io/docs/tasks/inject-data-application/
+          command:
+            - psql
+            - -qa
+            - --set=ON_ERROR_STOP=1
+            - --command
+            - |
+              DO $$$$
+              DECLARE
+                restored jsonb;
+              BEGIN
+                ASSERT pg_is_in_recovery(), 'expected replica';
+                -- only users with "pg_read_all_settings" role may examine "primary_conninfo"
+                -- ASSERT current_setting('primary_conninfo') <> '', 'expected streaming';
```

Comment:
💭 So could we test this with a `kubectl exec ... -- psql` command?

Reply:
We could. Should I add it?

There are a lot of "shell" steps in this test case, and sometimes it felt like cheating (?) ... All the data _could_ be checked by the superuser, but it seems more appropriate to come at it from the user account.

🤔 I'm thinking now it might be good to have a line in `14--lose-data` that shows the replica has lost data, too. (One more script that comes at it as the superuser.)

Reply:
Oh, that (showing the replica has lost data) makes a lot of sense; and yeah, as someone who does a lot of manual testing as both regular and super-user, I support checking as much from the regular user as possible. 

I'm not even sure I'd argue _for_ a special `kubectl exec` to check the `primary_conninfo`, I just thought I'd mentton it as it's pretty easy to do with KUTTL as a script (rather than job). That said, maybe it will save us a headache later if we make these tests  check as many elements as is reasonable? Any thoughts?

Reply:
Done. Added the check to `14--lose-data` and added a new `17--check-replication` that verifies the settings that are visible only to a superuser.